fabric-rendering-v1/src/client/java/net/fabricmc/fabric/mixin/client/rendering/shader/ShaderProgramImportProcessorMixin.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.client.rendering.shader;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.ModifyVariable;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.fabricmc.fabric.impl.client.rendering.FabricShaderProgram;
+
+/**
+ * Lets modded shaders {@code #moj_import} shaders from any namespace with the
+ * {@code <>} syntax.
+ */
+@Mixin(targets = "net.minecraft.client.gl.ShaderProgram$1")
+abstract class ShaderProgramImportProcessorMixin {
```

Comment:
How is this mixin supposed to work? Does it correctly take the path normalization into account?

![image](https://user-images.githubusercontent.com/13494793/220401116-d55363da-c605-4a4b-9769-3ca1761c6c55.png)

Reply:
Yes, it should work with normalisation as it modifies the normalised path. The original name without the directory `shaders/include/` is included for the second param of `FabricShaderProgram.rewriteAsId` (param 1 is the normalised path, i.e. the current value of `name`).

Reply:
We don't need to consider normalization since `\` is not a valid ID character. So this will be fine. :+1: